syntax = "proto3";

package envoy.extensions.matching.common_inputs.ssl.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.matching.common_inputs.ssl.v3";
option java_outer_classname = "SslInputsProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/matching/common_inputs/ssl/v3;sslv3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Common SSL matching inputs]

// List of comma-delimited URIs in the SAN field of the peer certificate for a downstream.
// [#extension: envoy.matching.inputs.uri_san]
message UriSanInput {
}

// List of comma-delimited DNS entries in the SAN field of the peer certificate for a downstream.
// [#extension: envoy.matching.inputs.dns_san]
message DnsSanInput {
}

// Input that matches the subject field of the peer certificate in RFC 2253 format for a
// downstream.
// [#extension: envoy.matching.inputs.subject]
message SubjectInput {
}
